crates/strategy/zksync/src/cheatcode/runner/mod.rs

```diff
@@ -825,6 +825,15 @@ impl CheatcodeInspectorStrategyExt for ZksyncCheatcodeInspectorStrategyRunner {
                     }
                 }
 
+                // We increase the max depth to account for the inspector calling initialize_interp
+                // for nested calls in evm. This is because the expected_revert max
+                // depth is set in this function. Given that we don't execute
+                // initialize_interp in zkEVM calls we need to account for it.
+                // For the expect_revert max depth check increasing it by one is sufficient.
+                if let Some(expected_revert) = &mut state.expected_revert {
+                    expected_revert.max_depth = ecx.journaled_state.depth() + 1;
```

Comment:
hmm what if the recorded `max_depth` from evm calls was higher than the current depth? wouldn't that overrride this? I see that the upstream logic does it with `max`

Reply:
Also, do we have this documented/tested anywhere?

Reply:
Yes, you are right I forgot to push the max. What do you mean with tested? The particular case of the max_depth being more?

Reply:
No, I was wondering if we have a test (or a doc) that we can reference which ensures that the `initialize_interp` is called each nested call